### PR TITLE
Feature/#12895 store forward mode

### PR DESCRIPTION
**Pull request by ovidiuvladpop**

### CODE REVIEW

File: ClearentIdtechIOSFramework/ClearentUI/ClearentUIManager.swift [hunk 1]
```diff
@@ -54,9 +59,10 @@ public final class ClearentUIManager: NSObject {
      * @param baseURL, the endpoint of the backend
      * @param apiKey, the API Key in order to use the API
      * @param publicKey, needed for the card reader initialisation
+     * @param enableOfflineMode, enables / disables the store & forward feature
      */
-    @objc public func updateWith(baseURL: String, apiKey: String, publicKey: String, enableEnhancedMessaging: Bool) {
-        clearentWrapper.updateWithInfo(baseURL: baseURL, publicKey: publicKey, apiKey: apiKey, enableEnhancedMessaging: enableEnhancedMessaging)
+    @objc public func updateWith(baseURL: String, apiKey: String, publicKey: String, enableEnhancedMessaging: Bool, enableOfflineMode: Bool) {
+        clearentWrapper.updateWithInfo(baseURL: baseURL, publicKey: publicKey, apiKey: apiKey, enableEnhancedMessaging: enableEnhancedMessaging, enableOfflineMode: enableOfflineMode)
```
Comment (carmenj) [inline ClearentIdtechIOSFramework/ClearentUI/ClearentUIManager.swift:65]: `enableOfflineMode` is already declared as a global variable which sets the one from `ClearentWrapper`, so I don't think changing this update method is needed.

File: ClearentIdtechIOSFramework/ClearentUI/Common/UIComponents/ManualEntryCardInfo/ClearentPaymentHeaderView.swift [hunk 2]
```diff
@@ -10,11 +10,20 @@ import Foundation
...
     override func configure() {
+        let titleLabel = UILabel()
         titleLabel.textColor = ClearentUIBrandConfigurator.shared.colorPalette.manualPaymentTitleColor
         titleLabel.font = ClearentUIBrandConfigurator.shared.fonts.paymentViewTitleLabelFont
         titleLabel.text = ClearentConstants.Localized.ManualEntry.header
+        titleLabel.textAlignment = .center
+        
+        stackView.addArrangedSubview(titleLabel)
+        
+        if ClearentWrapper.shared.enableOfflineMode && ClearentWrapper.shared.offlineModeState != .off {
```
Comment (carmenj) [inline ClearentIdtechIOSFramework/ClearentUI/Common/UIComponents/ManualEntryCardInfo/ClearentPaymentHeaderView.swift:25]: ideally, UI components should not have references to `ClearentWrapper`, so you could just pass a  parameter (subtitleText: String?) to a setup function for example.
 If this parameter has a value => insert `ClearentSubtitleLabel`.

File: ClearentIdtechIOSFramework/ClearentUI/PaymentFlowDataSource/FlowDataProvider.swift [hunk 3]
```diff
@@ -174,6 +186,17 @@ extension FlowDataProvider : ClearentWrapperProtocol {
...
     
+    func showEncryptionWarning() {
+        let items = [FlowDataItem(type: .graphicType, object: FlowGraphicType.warning),
+                     FlowDataItem(type: .title, object: ClearentConstants.Localized.OfflineMode.offlineModeEncryptionWarningMessage),
+                     FlowDataItem(type: .userAction, object: FlowButtonType.cancel)]
+        let flowFeedback = FlowDataFactory.component(with: .payment,
+                                                     type: .warning,
+                                                     readerInfo: fetchReaderInfo(),
+                                                     payload: items)
+        self.delegate?.didReceiveFlowFeedback(feedback: flowFeedback)
```
Comment (carmenj) [inline ClearentIdtechIOSFramework/ClearentUI/PaymentFlowDataSource/FlowDataProvider.swift:197]: I think `self` keyword is not needed

**Review by carmenj — CHANGES_REQUESTED** [review]

File: ClearentIdtechIOSFramework/ClearentWrapper/Wrapper/ClearentWrapperEnums.swift [hunk 4]
```diff
@@ -111,3 +114,11 @@ public enum UserAction: String, CaseIterable {
         return (text == ClearentConstants.Messaging.suppress) ? nil :UserAction(rawValue: text)
     }
 }
+
+public enum ReaderEncryption: Int {
+    case notEncrypted, emvEncryption, msrEncryption, fullyEncrypted
```
Comment (carmenj) [inline ClearentIdtechIOSFramework/ClearentWrapper/Wrapper/ClearentWrapperEnums.swift:119]: are all these cases used/needed?

Comment (ovidiuvladpop) [reply ClearentIdtechIOSFramework/ClearentWrapper/Wrapper/ClearentWrapperEnums.swift:119]: Removed all these cases and compared the response with an int value.

File: ClearentIdtechIOSFramework/ClearentUI/ClearentProcessingModalPresenter.swift [hunk 5]
```diff
@@ -74,6 +77,25 @@ class ClearentProcessingModalPresenter {
...
+        if let isReaderEncrypted = sdkWrapper.isReaderEncrypted(), useCardReaderPaymentMethod {
+            if !isReaderEncrypted {
+                sdkFeedbackProvider.showEncryptionWarning()
+            } else {
+                sdkFeedbackProvider.displayOfflineModeWarningMessage()
+            }
+        } else {
+            sdkFeedbackProvider.displayOfflineModeWarningMessage()
+        }
+    }
```
[the remark this answers is not in the document]
Comment (ovidiuvladpop) [reply ClearentIdtechIOSFramework/ClearentUI/ClearentProcessingModalPresenter.swift:97]: The suggestion from client was to display an error message (not implemented yet).

[the remark this answers is not in the document]
Comment (carmenj) [reply ClearentIdtechIOSFramework/ClearentUI/ClearentProcessingModalPresenter.swift:97]: please create a separate task for this

**Pull request merged**